File: snippets/fsharp/VS_Snippets_CLR_System/system.BitConverter.GetBytes.Others/FS/bytesbool.fs
module bytesbool

//<Snippet1>
open System

// Define Boolean true and false values.
let values = [ true; false ]

// Display the value and its corresponding byte array.
printfn "%10s%16s\n" "Boolean" "Bytes"

for value in values do
    let bytes = BitConverter.GetBytes value
    
    BitConverter.ToString bytes
    |> printfn "%10b%16s" value


// The example displays the following output:
//        Boolean           Bytes
//
//           true              01
//          false              00
//</Snippet1>

File: snippets/csharp/System/BitConverter/GetBytes/bytesbool.cs
//<Snippet1>
using System;

class GetBytesBooleanDemo
{
    public static void Main()
    {
        // Define Boolean true and false values.
        bool[] values = { true, false };

        // Display the value and its corresponding byte array.
        Console.WriteLine($"{"Boolean",10}{"Bytes",16}\n");
        foreach (bool value in values)
        {
            byte[] bytes = BitConverter.GetBytes(value);
            Console.WriteLine($"{value,10}{BitConverter.ToString(bytes),16}");
        }
    }
}
// The example displays the following output:
//        Boolean           Bytes
//
//           True              01
//          False              00
//</Snippet1>
